module/zfs/dsl_scan.c

```diff
@@ -835,6 +877,26 @@ dsl_scan_done(dsl_scan_t *scn, boolean_t complete, dmu_tx_t *tx)
 		 * Let the async thread assess this and handle the detach.
 		 */
 		spa_async_request(spa, SPA_ASYNC_RESILVER_DONE);
+
+		/*
+		 * If we have a deferred resilver waiting, start it
+		 * asynchronously now. dsl_scan_reset_deferred() may
+		 * update the config before the resilver can restart.
+		 * In the event of a crash during this period, the
+		 * spa loading code will find the drives that need
+		 * to be resilvered when the machine reboots and
+		 * start the resilver then.
+		 */
+		boolean_t should_resilver =
+		    dsl_scan_reset_deferred(spa->spa_root_vdev, tx);
+		if (should_resilver) {
+			zfs_dbgmsg("starting deferred resilver txg=%llu",
+			    (longlong_t)tx->tx_txg);
```

Comment:
This shouldn't be necessary - spa_history_log_internal() should also generate a dbgmsg (but feel free to double check this - code is in spa_history_log_sync()).